TrackletAlgorithm/VMRouter_parameters.h

```diff
@@ -0,0 +1,1565 @@
+#ifndef TrackletAlgorithm_VMRouter_parameters_h
+#define TrackletAlgorithm_VMRouter_parameters_h
+
+// Hardcoded number of memories and masks from the wiring.
+// NOTE: this file is incomplete. Please add the missing numbers
+
+// Enums used to get the correct parameters
+enum class phiRegions : char {A = 'A', B = 'B', C = 'C', D = 'D', E = 'E', F = 'F', G = 'G', H = 'H'};
+
+// The functions that returns the LUTs and parameters
+template<bool isLayer, int layer> const int* getPhiCorrTable();
+template<bool isLayer, int layer> const int* getRzBitsInnerTable();
+template<bool isLayer, int layer> const int* getRzBitsOverlapTable();
+template<bool isLayer, int layer> const int* getRzBitsOuterTable();
+template<bool isLayer, int layer, phiRegions phi> const int* getFineBinTable();
+template<bool isLayer, int layer, phiRegions phi, int size> const ap_uint<size>* getBendCutInnerTable();
+template<bool isLayer, int layer, phiRegions phi, int size> const ap_uint<size>* getBendCutOverlapTable();
+template<bool isLayer, int layer, phiRegions phi, int size> const ap_uint<size>* getBendCutOuterTable();
+
+template<bool isLayer, int layer, phiRegions phi> constexpr int getNumInputs();
+template<bool isLayer, int layer, phiRegions phi> constexpr int getNumInputsDisk2S();
+template<bool isLayer, int layer, phiRegions phi> constexpr int getNumASCopies();
+template<bool isLayer, int layer, phiRegions phi> constexpr int getNumTEICopies();
+template<bool isLayer, int layer, phiRegions phi> constexpr int getNumOLCopies();
+template<bool isLayer, int layer, phiRegions phi> constexpr int getNumTEOCopies();
+
+template<bool isLayer, int layer, phiRegions phi> constexpr int getBendCutTableSize();
```

Comment:
The `isLayer` parameter seems to be redundant, since it's completely determined by `layer`. Can it be removed?

Reply:
Ah I see the issue, and in fact, right now `TF::L1` and `TF::D1` are both equal to zero.

I would propose creating a new enum in the `TF` namespace in Constants.h that includes both layers and disks. Something like:
```c++
  enum layerDisk {L1 = 0, L2 = 1, L3 = 2, L4 = 3, L5 = 4, L6 = 5, D1 = 6, D2 = 7, D3 = 8, D4 = 9, D5 = 10};
```
We can keep the `layer` and `disk` enums for now, but eventually they would be phased out in favor of `layerDisk` (or whatever you choose to name it). This way, we eliminate the ambiguity between the disks and some of the layers, and you can also use `TF::layerDisk` as the type for the template parameter in your functions, instead of the generic `int`.

Reply:
I removed the layer and disk enums as we can't have multiple L1s etc. Only the MC was using them, so I fixed that. Hope that was okay :-)

Reply:
Perfect.

I realized this morning though that this change _should_ also break the TC and TP because `TF::D1`, `TF::D2`, etc. are used as indices into `zmean`. Unfortunately, this isn't reflected in the results of the CI tests because both modules have bugs in their respective test benches. I'll open a separate PR to fix these bugs, but I just added a commit to your branch that extends `zmean` and `rmean` so that the TC and TP work again.

Reply:
Ah okay thanks, I missed that